src/InstrumentationEngine.Lib/FileLoggerSink.cpp

```diff
@@ -80,8 +80,23 @@ HRESULT CFileLoggerSink::Reset(_In_ LoggingFlags defaultFlags, _Out_ LoggingFlag
 
         // The file name is empty if (a) the last character in the path is a path separator
         // or (b) if no file name is given e.g. the file starts with a period.
-        // Generate a file name if the file name is empty and the ext is either empty or "."
-        if (tsFile.empty() && (0 == tsExtension.compare(_T(".")) || 0 == tsExtension.compare(_T(""))))
+        // Generate a file name if we're given a directory.
+        bool isDirectory = false;
+        if (tsFile.empty())
+        {
+            isDirectory = (0 == tsExtension.compare(_T(".")) || 0 == tsExtension.compare(_T("")));
+        }
+        else 
+        {
+            // Check if fileName is actually a directory, meaning the path separator was left off
+            if ((GetFileAttributes(m_tsPathCandidate.c_str()) & FILE_ATTRIBUTE_DIRECTORY) != 0)
+            {
+                wcscat_s(wszDirectory, _MAX_DIR, wszFile);
```

Comment:
Is there a reason to not delete the string that exists on wszFile here? Looks like it'll exist in both wszDirectory and wszFile. Could that be what's causing the test errors?

Reply:
The test error was due to the FileAttributes returning `INVALID_FILE_ATTRIBUTES` which is -1 which succeeds our Directory flag check. Fixing.